### ISSUE TITLE
Typo: End == 2 ??

### ISSUE THREAD

**Issue by joelbrock** [CORE-POS/IS4C#993]
https://github.com/CORE-POS/IS4C/blob/f117b71ce81725155f0b59d853272c8876be63b3/pos/is4c-nf/gui-modules/ddd.php#L55

**Comment by gohanman** [comment]
I don't think so. I don't know *why* it's two but it appears to have been that way for a long time - e.g., 0.9.0 https://github.com/CORE-POS/IS4C/blob/0.9.0-2013.08.23/pos/is4c-nf/ajax-callbacks/ddd.php#L43

**Comment by joelbrock** [comment]
Huhn.  Well then, FWIW, if i change the 2 to a 1 then the DDD transaction ends successfully after admin approval.  With End == 2 the transaction remains onscreen following reason select and admin auth.

**Comment by gohanman** [comment]
Weird. It works fine for me with 2. And in any event the call to [TransRecord::finalizeTransaction](https://github.com/CORE-POS/IS4C/blob/f117b71ce81725155f0b59d853272c8876be63b3/pos/is4c-nf/gui-modules/ddd.php#L61) should clear the current transaction tables either way.

**Comment by joelbrock** [comment]
more details about the DDD behavior here: https://github.com/CORESupport/Alberta/issues/176#issuecomment-387826682

**Comment by gohanman** [comment]
Hmmm. I just tried the exact same sequence and I do not get the transaction-in-progress message. Is there something in `localtemptrans` at that point in their setup?

**Comment by gohanman** [comment]
That's thoroughly weird. I've tried it on 4 different lanes now and none of them show the problem. If `localtemptrans` is empty then `LastID` should be zero and the error in question shouldn't occur. I'm not sure what to do beyond littering `AjaxEnd.php` with extra lines logging to a file on the machine that's exhibiting this behavior to understand what's happening differently there. `LastID` should get reset to zero when `Database::getsubtotals` is called and I can't see any obvious way that call wouldn't happen.
